### DISCUSSION TITLE
Setting Item Priority and Item Section doesn't seem to do anything

### DISCUSSION THREAD

**Discussion by MikeBAlgWare** [olsh/todoist-net#52] [Q&A] [answered]
Hi, I'm trying to set a default section and priority on item updates.  Section seems to want the string name of the section, and priority seems to want a Todoist.Net.Models.Priority enum.  I set these things and I don't see the expected result in the synched task.  I am able to update content, description, and labels correctly.  Anybody else seeing this behavior?  Thanks Mike

**Comment by AhmedZaki99** [comment, marked as answer]
Hi Mike,
Currently, Todoist API doesn't support moving tasks to new sections with the [`item_update`](https://developer.todoist.com/sync/v9/#update-an-item) command, instead, they made another command just for moving tasks to new projects or sections called [`item_move`](https://developer.todoist.com/sync/v9/#move-an-item), which you can use by calling [`ItemsCommandService.MoveAsync`](https://github.com/olsh/todoist-net/blob/master/src/Todoist.Net/Services/ItemsCommandService.cs#L125) method.

You will find a note left in their documentation on [updating items](https://developer.todoist.com/sync/v9/#update-an-item):
> Please note that updating the parent, moving, completing or uncompleting tasks is not supported by item_update, more specific commands have to be used instead.

As for priorities, it's important to note that in Todoist API, values are reversed (1 to 4 means 4 to 1). You will find it described under the [Update an item section](https://developer.todoist.com/sync/v9/#update-an-item) as well:

> **priority** `Integer`
> The priority of the task (a number between 1 and 4, 4 for very urgent and 1 for natural).
> Note: Keep in mind that very urgent is the priority 1 on clients. So, p1 will return 4 in the API.

So, if you tried to update a task with the enum value `Priority.Priority1`, it will end up setting the priority of this task to the least urgent (default value), hence, nothing changes.

**Comment by MikeBAlgWare** [comment]
Ahmed thank you for all the pointers you have given me.  Both of these suggestions worked- I was thinking of "section" being an assignment and not a move.  As far as the reversed priority values go, that's just bonkers.  Thanks again for all your help - Mike
